services/QuillLMS/app/models/concerns/checkbox_callback.rb

```diff
@@ -4,11 +4,17 @@ module CheckboxCallback
   extend ActiveSupport::Concern
 
   def find_or_create_checkbox(name, user, activity_id=nil)
-    return unless Objective.find_by_name(name) && user
+    objective = Objective.find_by(name: name)
 
-    Checkbox.find_or_create_by(user_id: user.id, objective_id: Objective.find_by_name(name).id)
-    CheckboxAnalyticsWorker.perform_async(user.id, activity_id) if activity_id
-  rescue => e
-    puts "Race condition"
+    return unless objective && user
+
+    begin
+      Checkbox.transaction(requires_new: true) do
+        Checkbox.find_or_create_by!(user: user, objective: objective)
```

Comment:
Should this use [create_or_find_by](https://api.rubyonrails.org/classes/ActiveRecord/Relation.html#method-i-create_or_find_by-21). Seems to be closely mimicking that code.

![Screenshot 2022-12-14 at 10 28 12 AM](https://user-images.githubusercontent.com/1304933/207638510-94964725-5aab-4fbb-8dc9-f142e99faaf6.png)